lm75bd/lm75bd.c

```diff
@@ -26,8 +26,25 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 }
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
-  /* Implement this driver function */
-  
+  error_code_t errCode;  
+
+  uint8_t bufferTo[1] = {0x00U}; // Temperature Register
+  RETURN_IF_ERROR_CODE(i2cSendTo(devAddr, bufferTo, 1));
+
+  uint8_t bufferFrom[2] = {0x00U};
+  RETURN_IF_ERROR_CODE(i2cReceiveFrom(devAddr, bufferFrom, 2));
+
+  // The temp is only written in the first 11 bits
+  int16_t rawTemp = ((bufferFrom[0] << 8) | bufferFrom[1]) >> 5;
+ 
+
+  // Checking Two Complement, the next two lines were ChatGPT
+  // Checks the 10th bit for twos complement
+  if (rawTemp & 0x0400) {  
+    rawTemp |= 0xF800; // Correctly fills the twos complement in 16 bit int
+  }
```

Comment:
i think they are required, because im storing a 11 bit int in a int16_t in the following way 00000XXXXXXXXXXX. the code above changes it if it detects a two compliment bit, it gets changed to 11111XXXXXXXXXXX (the compliment in 16 bits).

I also tried running the test case without that code and it fails.